gdc_client/upload/parser.py

```diff
@@ -71,31 +70,31 @@ def config(parser):
     """ Configure a parser for upload.
     """
     func = partial(upload, parser)
-    parser.set_defaults(func=func)
+
+    conf = GDCClientUploadConfig()
+    upload_defaults = conf.to_dict()
+    upload_defaults['func'] = func
+
+    parser.set_defaults(**upload_defaults)
 
     parser.add_argument('--project-id', '-p', type=str,
                         help='The project ID that owns the file')
-    parser.add_argument('--path', '-f', metavar='path', default='',
+    parser.add_argument('--path', '-f', metavar='path', type=str,
```

Comment:
The defaults are going to come from the new config file right? That's neat.

Reply:
yes, this was the idea, but it seems like the config gets lost during packaging... need to figure out another way of loading the default one... might end up storing it in a dictionary it seems...

Reply:
ended up hard coding the defaults in code, because it's wonky to ensure that a configuration file is packaged properly.